src/utils/integration/flows.ts

```diff
@@ -50,3 +50,95 @@ export async function listIntegrationFlows(integrationId: string) {
 
   return flows;
 }
+
+export interface LogNode {
+  [index: string]: unknown;
+  timestamp: string;
+  severity: string;
+  message: string;
+}
+
+export interface FetchLogsResult {
+  logs: LogNode[];
+  cursor: string | undefined;
+  executionComplete?: boolean;
+}
+
+export async function getExecutionLogs(executionId: string, nextCursor?: string) {
+  return await gqlRequest({
+    document: gql`
+      query getExecutionLogs($executionId: ID!, $nextCursor: String) {
+        logs(
+          executionResult: $executionId
+          after: $nextCursor
+          orderBy: { field: TIMESTAMP, direction: ASC }
+        ) {
+          edges {
+            node {
+              timestamp
+              severity
+              message
+            }
+            cursor
+          }
+        }
+      }
+    `,
+    variables: {
+      executionId,
+      nextCursor,
+    },
+  });
+}
+
+export interface StepResultNode {
+  [index: string]: unknown;
+  stepName: string;
+  endedAt: string;
+  resultsUrl: string;
+}
+
+export async function getExecutionStepResults(executionId: string, nextCursor?: string) {
+  return await gqlRequest({
+    document: gql`
+      query getExecutionStepResults($executionId: ID!, $nextCursor: String) {
+        executionResult(id: $executionId) {
+          stepResults(after: $nextCursor, orderBy: { field: ENDED_AT, direction: ASC }) {
+            edges {
+              node {
+                stepName
+                endedAt
+                resultsUrl
+              }
+              cursor
+            }
+            totalCount
+          }
+        }
+      }
+    `,
+    variables: {
+      executionId,
+      nextCursor,
+    },
+  });
+}
+
+export async function isCniExecutionComplete(executionId: string) {
+  const result = await gqlRequest({
+    document: gql`
+      query isCniExecutionComplete($executionId: ID!) {
+        executionResult(id: $executionId) {
+          stepResults {
+            totalCount
+          }
+        }
+      }
+    `,
+    variables: {
+      executionId,
+    },
+  });
+
+  return result.executionResult.stepResults.totalCount === 2;
```

Comment:
Question on this, if we run a normal execution how does this determine if its a CNI integration (with two steps) vs a normal integration? Does `isCniExecutionComplete` run on normal executions too.

Reply:
We have no way of currently knowing if an execution is for a CNI or not unless the user provides an integrationId that we can query against (e.g. `integration.isCodeNative`). If the user is invoking the flow directly via an invocation URL then we really have no way to tell.

As opposed to removing the `--cni-auto-end` flag entirely, I just left it in there with the expectation that things will just not work as expected with low-code flows and figured that would be acceptable given that the flag description and flag itself says that it's for CNI only.